trackintel/preprocessing/staypoints.py

```diff
@@ -180,20 +180,17 @@ def generate_locations(
     # merge non_activities back if "activities_only" flag is set
     sp = pd.concat([sp, non_activities])
 
-    if len(locs) > 0:
-        locs.as_locations  # empty location is not valid
-    else:
-        warnings.warn("No locations can be generated, returning empty locs.")
-
     ## dtype consistency
     # locs id (generated by this function) should be int64
     locs.index = locs.index.astype("int64")
     # location_id of staypoints can only be in Int64 (missing values)
     sp["location_id"] = sp["location_id"].astype("Int64")
     # user_id of locs should be the same as sp
     locs["user_id"] = locs["user_id"].astype(sp["user_id"].dtype)
-
-    return sp, locs
+    if len(locs) <= 0:
+        warnings.warn("No locations can be generated, returning empty locs.")
+        return sp, locs
+    return Staypoints(sp), Locations(locs)
```

Comment:
See above comments; either keep the line as it is or change it to `return sp, Locations(locs)`

Reply:
Here `sp` is created as a new object instance. So I used `Staypoints`. But I agree that this should be consistent with the other functions.

Reply:
okay, I see a line `sp= gpd.GeoDataFrame()` above in our code, so its fine to keep it like this